web/client/utils/cesium/WMSUtils.js

```diff
@@ -6,8 +6,76 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-const WMSUtils = {
-    PARAM_OPTIONS: ["layers", "styles", "style", "format", "transparent", "version", "tiled", "opacity", "zindex", "srs", "singletile", "_v_", "filterobj" ]
+import { getAuthenticationHeaders } from "../SecurityUtils";
+import { getProxyUrl, needProxy } from "../ProxyUtils";
+import ConfigUtils from "../ConfigUtils";
+
+const PARAM_OPTIONS = ["layers", "styles", "style", "format", "transparent", "version", "tiled", "opacity", "zindex", "srs", "singletile", "_v_", "filterobj" ];
+
+
+function splitUrl(originalUrl) {
+    let url = originalUrl;
+    let queryString = "";
+    if (originalUrl.indexOf('?') !== -1) {
+        url = originalUrl.substring(0, originalUrl.indexOf('?') + 1);
+        if (originalUrl.indexOf('%') !== -1) {
+            url = decodeURIComponent(url);
+        }
+        queryString = originalUrl.substring(originalUrl.indexOf('?') + 1);
+    }
+    return {url, queryString};
+}
+
+function WMSProxy(proxy) {
+    this.proxy = proxy;
+}
+
+WMSProxy.prototype.getURL = function(resource) {
+    let {url, queryString} = splitUrl(resource);
+    return getProxyUrl() + encodeURIComponent(url + queryString);
+};
+
+function NoProxy() {}
+
+NoProxy.prototype.getURL = function(resource) {
+    const { url, queryString } = splitUrl(resource);
+    return url + queryString;
+};
+
+// Check and apply proxy to source url
+export const getProxy = (options) => {
+    let proxyUrl = ConfigUtils.getProxyUrl({});
+    let proxy;
+    if (proxyUrl) {
+        proxy = options.noCors || needProxy(options.url);
+    }
+    return proxy ? new WMSProxy(proxyUrl) : new NoProxy();
 };
 
-module.exports = WMSUtils;
+/**
+ * Generate cesium BIL option for BILTerrainProvider from wms layer option
+ * @param {object} options
+ * @returns {object} converted BIL options
+ */
+export const wmsToCesiumOptionsBIL = (options) => {
+    let url = options.url;
+    const headers = getAuthenticationHeaders(url, options.securityToken);
+    return {
+        url,
+        headers,
+        proxy: getProxy(options),
+        littleEndian: options.littleendian || false,
+        layerName: options.name,
+        version: options.version,
+        fixedHeight: options.fixedHeight,
+        fixedWidth: options.fixedWidth,
+        crs: options.crs, // Support only CRS:84 | EPSG:4326 | EPSG:3857 | OSGEO:41001
+        format: options.format // MapStore only supports WMS "image/bil"
```

Comment:
I think we should highlight this option change in the migration guideline because older configuration uses `application/bil16` and now `image/bil`.
The other option is to hardcode here "image/bil" because at the end we can only use that for the moment